%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails

% Modifications for comments
% \usepackage[paperwidth=275.9mm, paperheight=279.4mm]{geometry} %regular letter size is 215.9 wide by 279.44 long
% \setlength{\evensidemargin}{95mm}
%\usepackage[dvipsnames,svgnames,x11names]{xcolor}
\usepackage[markup=underlined]{changes}
\usepackage[T1]{fontenc}

\definechangesauthor[color=red]{Corrinne}
\definechangesauthor[color=blue]{Justin}
%%% Alternative definition to have the remarks
%%% in the margins instead of footnotes
\usepackage{todonotes}
\setlength{\marginparwidth}{3cm}
\makeatletter
\setremarkmarkup{\todo[color=Changes@Color#1!20,size=\scriptsize]{#1: #2}}
\makeatother

%% Rather hacky definition of a plain remark/note
%% by riding on \added
\newcommand{\note}[2][]{\added[id=#1,remark={#2}]{}}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\title{Thirteen genome sequences representing the entire subgenus Houzingenia (Gossypium): insights into evolution of the New World diploid cottons}
\title{Thirteen genomes}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   corref={aff1},                       % id of corresponding address, if any
   noteref={n1},                        % id's of article notes, if any
   email={corrinne@iastate.edu}   % email address
]{\inits{CE}\fnm{Corrinne E} \snm{Grover}}
\author[
   addressref={aff2},
   email={maa146@igbb.msstate.edu}
]{\inits{MA}\fnm{Mark A} \snm{Arick II}}
\author[
addressref={aff1},
email={jconover@iastate.edu}
]{\inits{JC}\fnm{Justin C} \snm{Conover}}
\author[
addressref={aff3},
email={Jodi.Scheffler@ARS.USDA.GOV}
]{\inits{JA}\fnm{Jodi A} \snm{Scheffler}}
\author[
addressref={aff2},
email={william.s.sanders@gmail.com}
]{\inits{WS}\fnm{William S} \snm{Sanders}}
\author[
addressref={aff2},
email={peterson@IGBB.msstate.edu}
]{\inits{DG}\fnm{Daniel G} \snm{Peterson}}
\author[
addressref={aff3},
email={Brian.Scheffler@ARS.USDA.GOV}
]{\inits{BE}\fnm{Brian E} \snm{Scheffler}}
\author[
addressref={aff1},
email={jfw@iastate.edu}
]{\inits{JF}\fnm{Jonathan F} \snm{Wendel}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Ecology, Evolution, and Organismal Biology, Iowa State University}, % university, etc
  %\street{2200 Osborn Dr},                     %
  %\postcode{50011}                                % post or zip code
  \city{Ames, IA 50011},                              % city
  \cny{USA}                                    % country
}
\address[id=aff2]{%
  \orgname{Institute for Genomics, Biocomputing, and Biotechnology, Mississippi State University},
  %\street{Research Blvd},
  %\postcode{39762}
  \city{Mississippie State, MS 39762},
  \cny{USA}
}
\address[id=aff3]{%
	\orgname{Jamie Whitten Delta States Research Center, USDA-ARS},
	%\street{Research Blvd},
	%\postcode{38776}
	\city{Stoneville, MS 38776},
	\cny{USA}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{Background} %if any
Text for this section.

\parttitle{Results} %if any
Text for this section.

\parttitle{Conclusions} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{genome sequence}
\kwd{cotton}
\kwd{\textit{Gossypium}}
\kwd{molecular evolution}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
The American diploid “D-genome” cottons (subgenus \textit{Houzingenia}) comprise a monophyletic clade of cytogenetically and morphologically distinct species largely distributed from Southwest Mexico to Arizona, with additional disjunct species distributions in Peru and the Galapagos Islands \note[Corrinne]{which citations here, Endrizzi et al 1985}. Among the 13 species currently included in the D-genome \note[Corrinne]{although see Ulloa 2013} are\textit{G. harknessii} (D2-2), an important species for cytoplasmic male sterility in cotton, and \textit{G. raimondii} (D5), the model diploid progenitor to wild and domesticated allopolyploid cotton \note[Corrinne]{Endrizzi 1985, others?}. The close relationship of \textit{Houzingenia} species to the agronomically important polyploids, combined with the relative ease of sampling this subgenus for early cotton taxonomists, facilitated much of the current understanding of the relationships among D-genome species. 

These early taxonomists divided subgenus \textit{Houzingenia} into two sections and six subsections, whose species alliances have largely been retained by subsequent phylogenetic studies \note[Corrinne]{Cronn et al., 1996; Seelanan et al., 1997; Small and Wendel, 2000b; Wendel and Albert, 1992; Wendel et al., 1995b; alvarez 2005}. Several molecular datasets have been used to evaluate these relationships, including chloroplast restriction sites \note[Corrinne]{citation}; simple sequence repeat (SSR) and expressed sequence tag (EST)-SSR markers \note[Corrinne]{citation}; random amplified polymorphic DNA (RAPD) markers \note[Corrinne]{citation}; internal transcribed sequences (ITS) \note[Corrinne]{citation}; and few single-copy nuclear genes \note[Corrinne]{citation}. Relationships among the six subsections, however, remain unclear despite numerous, and often conflicting, studies \note[Corrinne]{Cronn et al., 1996; Liu et al., 2001b; Small and Wendel, 2000b more citations}. Determining the closest living relative of the D-genome ancestor to the polyploid, however, has been met with greater success. Early morphological and cytogenetic comparisons using intergenomic hybrids quickly identified \textit{G. raimondii} as the closest living relative to the D-genome ancestor of polyploid cotton species \note[Corrinne]{Stephens, 1944b Hutchinson et al., 1945, 1947 ; Gerstel and Phillips, 1958; Phillips, 1964; reviewed in Wendel and Cronn 2003}. Subsequent analyses have largely supported this observation (Abdalla et al., 2001; cronn 1999, liu 2001, Cronn et al., 1996 Seelanan et al., 1997 Small et al., 1998; Small and Wendel, 2000a,b), with few conflicts \note[Corrinne]{see, however Wendel et al., 1995b; I know there are others…. What are they?}.

A secondary outcome of this research has been the elucidation of multiple instances of hybridization among D-genome (i.e, \textit{Houzingenia}) species \note[Corrinne]{cite Cryptic Trysts}, and, in one remarkable case (i.e., \textit{G. gossypioides}), between a \textit{Houzingenia} species and another, geographically isolated subgenus from Africa (either A-, B-, or, F-genome \note[Corrinne]{cite cryptic Trysts again}). Notably, \textit{G. gossypioides} is multiply introgressant, with subsequent hybridization to a member of the \textit{G. raimondii} lineage resulting in chloroplast, if not further (and cryptic), nuclear introgression (Cronn 2003, cryptic trysts). Cytoplasmic introgression, and possibly cryptic nuclear, is also present in some populations of \textit{G. aridum}, i.e., the Mexican Colima populations; \textit{G. aridum} accessions derived from this location possess a \textit{G. davidsonii}- or \textit{G. klotzschianum}-like cytoplasm.

Modest attempts at understanding the evolution of the repetitive fraction of this genus support the inference of African introgression in \textit{G. gossypioides} \note[Corrinne]{Zhao 1998}; however, little else is understood with respect to the evolution of the non-genic fraction of \textit{Houzingenia}. The D-genome cottons possess the smallest genome sizes in the genus, ranging only ~1.11 fold, from 841 Mb – 934 Mb.  Notably, the distribution of genome sizes among the subsections suggests that this subgenus has experienced differential growth and/or reduction in genome size among species \note[Corrinne]{need a figure}; however, the patterns of sequence gain and loss have not been characterized for the subgenus. While the differences in genome size are not dramatic, there is evidence that the transposable element types which have accumulated in \textit{G. raimondii} are different than those that have achieved higher copy numbers than the remainder of the genus \note[Corrinne]{Jennifer, A domestication, kokia}. Furthermore, research comparing the two sister genera to cotton (i.e., \textit{Kokia} and \textit{Gossypioides}; \note[Corrinne]{cite Kokia paper}) reveals that their apparently static genome sizes belies both gain and loss of repetitive sequence, a result similar to that of the extant members of the A-genome (subgenus \textit{Gossypium}), whose small change in genome size (~1.05X) masks differences in element accumulation \note[Corrinne]{cite Agenome domestication}.

Modern sequencing techniques make it easy to produce a substantial amount of genomic sequencing suitable for addressing these basic questions in a more genomically comprehensive manner. Here we use modest coverage Illumina sequencing to present an in-depth view of the evolution subgenus Houzingenia, the cotton D-genome clade. We leverage newly generated genome and plastome sequences, representing the first for many species, to address questions surrounding genome evolution in a monophyletic group of closely related species. We characterize the patterns of molecular evolution of both genes and repetitive sequences to provide insight into the pace and pattern of evolution in this subgenus. For the first time, intergenic regions are evaluated to characterize the amount of divergence outside of genes, and due to indels or single-nucleotide polymorphisms (SNPs). Finally, we revisit the phylogeny of the D-genome, both adding additional insight into the relationships among species using hundreds of nuclear genes, as well as addressing questions regarding sequence gain and loss among closely related species. The genome characterized here not only provides insight into molecular evolution on a relatively recent timeframe, but it also provides resources for comparative research and the cotton community at large.

%\cite{koon,oreg,khar,zvai,xjon,schn,pond,smith,marg,hunn,advi,koha,mouse}

\section*{Results}
\subsection*{Genome assemblies and annotation}
Approximately 20X raw coverage libraries \note[Corrinne]{update numbers} were sequenced for at least one representative of each D-genome species (Table WHATEVER), resulting in an average of 54 M \note[Corrinne]{update numbers} reads per accession. Quality filters further reduced the number of reads per sample to an average of 38 M (range: 1,824 – 260 M\note[Corrinne]{update numbers}), representing an average of 14X coverage per sample\note[Corrinne]{update numbers}. Two accessions, \textit{G. thurberi} acc. 2 and G. \textit{harknessii} acc. 7, had few retained reads and therefore were retained solely for repetitive analysis.  The remaining accessions were assembled via ABySS using multiple k-mer values (see methods), and the assembly with the greatest E-size (citation) was selected for each species. E-size is an alternative assembly statistic implemented by the Genome Assembly Gold-standard Evaluations (GAGE) study to evaluate the completeness of the assembled gene space by considering the expected contig size for a randomly selected base. Here, the assembled E-sizes range from 1,066 bp in (D10) \note[Corrinne]{update now that D10 has better sequence} to 10,495 bp in (D11), with an average E-size 5,356 bp (Table Assembly Stats; Figure Assembly Stats). These two species also respectively had the smallest and largest maximum contig size, i.e., 7,784 bp for (D10) \note[Corrinne]{update} and 121,185 bp for (D11) (average = 57,076 bp \note[Corrinne]{update numbers}; Table Assembly Stats). Given an average gene size in the published \textit{G. raimondii genome} of 2,583 bp, these metrics suggest that most genes will be assembled in these genomes. Indeed, in comparison to the published gene predictions for \note[Corrinne]{update numbers}, over 94\% of genes were recovered from at least 75\% of the accessions, where a gene was considered present if more than 67\% of the gene was recovered from that accession. BUSCO analysis of the assembled genomes suggest general completeness of the gene space, with an average of \textbf{HOWMANY} complete BUSCOs recovered from each accession.

Chloroplast reads were also recovered from the raw reads, representing an average of XX\% \note[Corrinne]{include numbers} of the filtered sequencing reads.  These were independently assembled (HOW). The average assembled chloroplast genome size was \textbf{HOWBIG}, which is comparable to previously published cotton chloroplast genomes (lots of citations here), and these contained an average of \textbf{HOWMANY} \% ambiguity \note[Corrinne]{\% N} (Table Assembly Stats).


\subsection*{Gene evolution stuff}
Text for this sub-heading \ldots

\subsection*{Phylogenetics}
Text for this sub-heading \ldots

\subsection*{Transposable element characterization}
Similar to previous reports, repetitive DNAs contribute roughly half of the total genome sequence for all species in subgenus \textit{Houzingenia}, from an average of 44.5\% in D7 to 52\% in \textit{G. anomalum D2-1}. Like most flowering plants, a vast majority of this sequence is due to the occupation of class II \textit{gypsy} elements, which comprise 

Multi-dimensional transposable element profile visualization using both log transformed and percent-genome size standardized counts showed considerable overlap among species, and even among subsections (Figure Ordination). Multivariate t-distribution confidence ellipses for each subsection overlap at least one other. Even those subsections where insufficient sampling precludes the generation of a confidence ellipse (i.e., Selera and Integrifolia), the plotted data points are contained within the occupied space of another subsection. Selera, for example, is contained within the confidence ellipse for both Caducibractaea and Houzingenia; Integrifolia is within Houzingenia and Austroamericana. PCA visualization of the same data suggests that 31.6\% and 13.8\% of the variability can be explained by the first two components; however, to formerly compare the overlap in repetitive profiles among subsections, we performed a Procrustes ANOVA with complex linear models, as implemented in the R package \{geomorph\}. For this analysis, we compared each subsection using all representatives of that subsection as indicators of variance. Few comparisons showed statistically significant differences, with the patterns of repetitive abundance differing only in Austroamericana versus both Caducibractaea and Erioxylum (p<0.05). Interestingly, the variation in repetitive elements found in monotypic Selera, i.e., \textit{G. gossypioides}, was not distinct from the remainder of subgenus \textit{Houzingenia}. This stands in contrast to previous reports (\textbf{cite these}), which noted a relative abundance of repeats derived from "African cottons" (here represented by subgenera \textit{Gossypium} and \textit{Longiloba}, i.e., A- and F-genome species). This result is further apparent when including the African subgenera in the ordination (Figure African Ordination); that is, \textit{G. gossypioides} is clearly lumped with the other \textit{Houzingenia} species. 

\subsection*{Ancestral state reconstructions}
Text for this sub-heading \ldots

\section*{Discussion}
\subsection*{Discussion subheading}

\section*{Conclusions}

\section*{Methods}
\subsection*{Sequence generation and initial processing}
DNA was extracted from (LEAVES) using (WHAT KIT), and sent to (WHERE) for library construction and sequencing.  Sequencing was completed on the Illumina (WHAT MACHINE) using (WHICH SEQUENCING). The data were trimmed and filtered with Trimmomatic v0.32 \note[Corrinne]{citation} with the following options : (1) sequence adapter removal, (2) removal of leading and/or trailing bases when the quality score (Q) <28, (3) removal of bases after average Q <28 (8 nt window) or single base quality <10, and (4) removal of reads < 85 nt. Detailed parameters can be found at https://github.com/williamssanders/D\_Cottons\_USDA. \note[Corrinne]{Let's port this repo to a lab site after and give the new url}

\subsection*{Genome assembly and annotation}
The trimmed data was independently assembled for each species via ABySS v2.0.1 \note[Corrinne]{citation}, using every 5th kmer value from 40 through 100. A single assembly with the highest E-size  (an alternative statistic to N50; \note[Corrinne]{citation Salzberg 2011}) was selected for each species and subsequently annotated with MAKER v2.31.6 \note[Corrinne]{citation} using evidence from: (1) the NCBI \textit{G. raimondii} EST database \note[Corrinne]{citation}, (2) \textit{G. raimondii} reference genome predicted proteins, as hosted by CottonGen.org \note[Corrinne]{citation}, and (3) three \textit{ab initio} gene prediction programs, i.e. Genemark v4.30 \note[Corrinne]{citation}, SNAP v2013-11-29 \note[Corrinne]{citation}, and Augustus v3.0.3 \note[Corrinne]{citation}. Both the SNAP and Augustus models were trained using BUSCO v2.0 \note[Corrinne]{citation}.    

\subsection*{Gene stuff}
Gene orthology and family designations were determined via OrthoFinder \note[Corrinne]{citation}...

\subsection*{Phylogenetics and ancestral state reconstruction}
Trimmed reads from the genome assembly were mapped against the \textit{G. raimondii} reference sequence (cite Paterson 2012) using BWA v0.7.10 (citation), post-processed with samtools (which version) (citation), and individual genes were independently assembled for each species/accession via BamBam v 1.4  (citation) in conjunction with the \textit{G. raimondii} reference annotation (cite Paterson 2012).  Alignments were pruned for genes and/or alignment positions with insufficient coverage, i.e., too many ambiguous bases, using filter\_alignments (https://github.com/williamssanders/D\_Cottons\_USDA/). Parameters were set  to remove sequences with more than 10\% ambiguous bases within species and to remove aligned positions with more than 10\% ambiguity among species. Genes were additionally filtered by length, to retain only those genes between a minimum of 500 bp (cite Mirarab 2016) and a maximum of 4051 bp, the latter of which represents the \textit{G. raimondii} genome-wide mean plus three standard deviations. Only those genes with a minimum of one accession per species were retained for phylogenetic and molecular analyses. 

Species trees were estimated from individual gene trees via SNaQ (citation) and MP-EST (citation) using Bayesian and Maximum Likelihood analyses, respectively. Bayesian analyses were generated using MrBayes v (which version) (cite Ronquist and Huelsenbeck 2003) under GTR gamma with the following parameters: four runs with four chains for 1 million generations and using a burn-in fraction of 25\%. Concordance among individual gene trees was assessed via BUCKy (Ané et al., 2007; Larget et al., 2010) with 3 runs, each with 4 chains and 1 million iterations, and default parameters. Quartet MaxCut was used to estimate the starting tree, and SNaQ was run using this starting tree and the concordance factors estimated by BUCKy. Visualization of networks\ldots

Maximum likelihood (ML) analyses were performed using RaxML v(which version) (cite Stamatakis 2014) using the basic general time reversible model with gamma distribution (GTRGAMMA), 10000 alternative runs on distinct starting trees, and rapid bootstrapping with consensus tree generation. The ML trees were rooted with a member of subgenus \textit{Longiloba}, \textit{G. longicalyx} (African F-genome). MP-EST (citation) was used to estimate the species tree from the population of gene trees.  Visualized how\ldots

Measures of molecular evolution were all calculated in R v(which version)(citation). Species divergence estimates were calculated via the \{chronoMPL\} package (citation), using the (which time estimates?) (citation). Trees were visualized using the \{ape\} package (citation). Ancestral state reconstructions for genome size were completed using {fastAnc}. Indels and SNPs were characterized among \textit{Houzingenia} using the Genome Analysis ToolKit (gatk) and the \textit{G. raimondii} reference sequence (Paterson citation). Distance measures of aligned intergenic regions were estimated via {ape}, and indels were characterized by (???). 

\subsection*{Repetitive characterization}
Reads from only one of the paired-end files (i.e., R1) were filtered and trimmed via Trimmomatic version 0.33 \cite{Bolger2014} to a uniform 95nt (https://github.com/williamssanders/D\_Cottons\_USDA), and then randomly subsampled to represent a 1\% genome size equivalent (GSE) for each individual \cite{Hendrix2005, Wendel2002}. These 1\% GSEs were combined as input into the RepeatExplorer pipeline \cite{Novak2013, Novak2010}, which has been successfully used to profile genomic repeats using low-coverage, short read sequencing. Only clusters which contain at least 0.01\% of the total input sequences (i.e., XXX \note[Corrinne]{numbers please}reads from a total input of X,XXX,XXX reads) were retained for annotation as per (KOKIA CITATION), which uses the RepeatExplorer implementation of RepeatMasker \cite{Smit2015} and a custom cotton-enriched repeat library. Genome occupation of each broad repeat type was calculated (in megabases; Mb) for each genome/accession based on the 1\% genome representation of the sample and the standardized read length of 95 nt. 

Broad patterns of repeat occupation per genome were determined by using the abundance of each cluster in a multivariate dataset. Initial visualization of the data was conducted using both calculated in R \note[Corrinne]{citation} using both Principle Coordinate Analysis on read counts, either log normalized (to compare overall patterns of repeats) or normalized by genome size (to compare proportional cluster occupation). Differential abundance in cluster occupation was iteratively calculated in increasing phylogenetic depths to understand the evolution of repeat types throughout the evolution of the subgenus; that is, differentially abundant clusters were determined (1) within species, (2) between sister taxa, and (3) between deeper phylogenetic nodes. For each cluster, the ancestral state was reconstructed and used for comparison in the next analysis. Ancestral state reconstructions were completed using \{fastAnc\} for reconstruction and the fitContinuous function of {Geiger} for visualization. All analyses are available at (https://github.com/williamssanders/D\_Cottons\_USDA).

\subsection*{Repeat heterogeneity and relative age}
Relative cluster age was approximated using the among-read divergence profile of each cluster, as previously used for Fritillaria \cite{Kelly2015}, dandelion \cite{Ferreira2016}, and \textit{Kokia}/\textit{Gossypioides}, sister outgroup genera to \textit{Gossypium}. Briefly, cluster-by-cluster all-versus-all BLASTn \cite{Boratyn2013, Altshul1990} searches were conducted using the same BLAST parameters implemented in RepeatExplorer. A pairwise percent identity histogram was generated for each cluster, and regression models were used to describe the trend (i.e., biased toward high-identity, “young” or lower-identity, “older” element reads) using Bayesian Information Criterion \cite{Schwarz1978} to select the model with the most confidence; specific parameters can be found in (KOKIA MANUSCRIPT) and at https://github.com/williamssanders/D\_Cottons\_USDA. The read similarity profile was automatically evaluated for each cluster to determine if the reads trend toward highly similar “young” or more divergent “older” reads. These profiles generally consist of six different trends: (1) positive linear regression ("young"); (2) absence of linear regression ("old"); (3) negative linear regression ("old"); (4) positive quadratic vertical parabola, trend described by right-side of vertex ("young"); (4b) positive quadratic vertical parabola, trend described by left-side of vertex ("old"); (5) negative quadratic vertical parabola, trend described by right-side of vertex ("old"); and (6) negative quadratic vertical parabola, trend described by left-side of vertex and vertex at >99\% pairwise-identity ("old"; Figure WHATEVER). We note that young" and "old" are relative designations and not indicative of absolute age. 

%\nocite{oreg,schn,pond,smith,marg,hunn,advi,koha,mouse}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  We would like to thank....
  
  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      A short description of the figure content
      should go here.}
      \end{figure}

\begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      Figure legend text.}
      \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\begin{table}[h!]
\caption{Sample table title. This is where the description of the table should go.}
      \begin{tabular}{cccc}
        \hline
           & B1  &B2   & B3\\ \hline
        A1 & 0.1 & 0.2 & 0.3\\
        A2 & ... & ..  & .\\
        A3 & ..  & .   & .\\ \hline
      \end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
