%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails

% Modifications for comments
% \usepackage[paperwidth=275.9mm, paperheight=279.4mm]{geometry} %regular letter size is 215.9 wide by 279.44 long
% \setlength{\evensidemargin}{95mm}
%\usepackage[dvipsnames,svgnames,x11names]{xcolor}
\usepackage[markup=underlined]{changes}
\usepackage[T1]{fontenc}

\definechangesauthor[color=red]{Corrinne}
\definechangesauthor[color=blue]{Justin}
%%% Alternative definition to have the remarks
%%% in the margins instead of footnotes
\usepackage{todonotes}
\setlength{\marginparwidth}{3cm}
\makeatletter
\setremarkmarkup{\todo[color=Changes@Color#1!20,size=\scriptsize]{#1: #2}}
\makeatother

%% Rather hacky definition of a plain remark/note
%% by riding on \added
\newcommand{\note}[2][]{\added[id=#1,remark={#2}]{}}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\title{Thirteen genome sequences representing the entire subgenus Houzingenia (Gossypium): insights into evolution of the New World diploid cottons}
\title{Thirteen genomes}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   corref={aff1},                       % id of corresponding address, if any
   noteref={n1},                        % id's of article notes, if any
   email={corrinne@iastate.edu}   % email address
]{\inits{CE}\fnm{Corrinne E} \snm{Grover}}
\author[
   addressref={aff2},
   email={maa146@igbb.msstate.edu}
]{\inits{MA}\fnm{Mark A} \snm{Arick II}}
\author[
addressref={aff1},
email={jconover@iastate.edu}
]{\inits{JC}\fnm{Justin C} \snm{Conover}}
\author[
addressref={aff3},
email={Jodi.Scheffler@ARS.USDA.GOV}
]{\inits{JA}\fnm{Jodi A} \snm{Scheffler}}
\author[
addressref={aff2},
email={william.s.sanders@gmail.com}
]{\inits{WS}\fnm{William S} \snm{Sanders}}
\author[
addressref={aff2},
email={peterson@IGBB.msstate.edu}
]{\inits{DG}\fnm{Daniel G} \snm{Peterson}}
\author[
addressref={aff3},
email={Brian.Scheffler@ARS.USDA.GOV}
]{\inits{BE}\fnm{Brian E} \snm{Scheffler}}
\author[
addressref={aff1},
email={jfw@iastate.edu}
]{\inits{JF}\fnm{Jonathan F} \snm{Wendel}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Ecology, Evolution, and Organismal Biology, Iowa State University}, % university, etc
  %\street{2200 Osborn Dr},                     %
  %\postcode{50011}                                % post or zip code
  \city{Ames, IA 50011},                              % city
  \cny{USA}                                    % country
}
\address[id=aff2]{%
  \orgname{Institute for Genomics, Biocomputing, and Biotechnology, Mississippi State University},
  %\street{Research Blvd},
  %\postcode{39762}
  \city{Mississippie State, MS 39762},
  \cny{USA}
}
\address[id=aff3]{%
	\orgname{Jamie Whitten Delta States Research Center, USDA-ARS},
	%\street{Research Blvd},
	%\postcode{38776}
	\city{Stoneville, MS 38776},
	\cny{USA}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{Background} %if any
Text for this section.

\parttitle{Results} %if any
Text for this section.

\parttitle{Conclusions} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{genome sequence}
\kwd{cotton}
\kwd{\textit{Gossypium}}
\kwd{molecular evolution}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
The American diploid “D-genome” cottons (subgenus \textit{Houzingenia}) comprise a monophyletic clade of cytogenetically and morphologically distinct species largely distributed from Southwest Mexico to Arizona, with additional disjunct species distributions in Peru and the Galapagos Islands \note[Corrinne]{which citations here, Endrizzi et al 1985}. Among the 13 species currently included in the D-genome \note[Corrinne]{although see Ulloa 2013} are\textit{G. harknessii} (D2-2), an important species for cytoplasmic male sterility in cotton, and \textit{G. raimondii} (D5), the model diploid progenitor to wild and domesticated allopolyploid cotton \note[Corrinne]{Endrizzi 1985, others?}. The close relationship of \textit{Houzingenia} species to the agronomically important polyploids, combined with the relative ease of sampling this subgenus for early cotton taxonomists, facilitated much of the current understanding of the relationships among D-genome species. 

These early taxonomists divided subgenus \textit{Houzingenia} into two sections and six subsections, whose species alliances have largely been retained by subsequent phylogenetic studies \note[Corrinne]{Cronn et al., 1996; Seelanan et al., 1997; Small and Wendel, 2000b; Wendel and Albert, 1992; Wendel et al., 1995b; alvarez 2005}. Several molecular datasets have been used to evaluate these relationships, including chloroplast restriction sites \note[Corrinne]{citation}; simple sequence repeat (SSR) and expressed sequence tag (EST)-SSR markers \note[Corrinne]{citation}; random amplified polymorphic DNA (RAPD) markers \note[Corrinne]{citation}; internal transcribed sequences (ITS) \note[Corrinne]{citation}; and few single-copy nuclear genes \note[Corrinne]{citation}. Relationships among the six subsections, however, remain unclear despite numerous, and often conflicting, studies \note[Corrinne]{Cronn et al., 1996; Liu et al., 2001b; Small and Wendel, 2000b more citations}. Determining the closest living relative of the D-genome ancestor to the polyploid, however, has been met with greater success. Early morphological and cytogenetic comparisons using intergenomic hybrids quickly identified \textit{G. raimondii} as the closest living relative to the D-genome ancestor of polyploid cotton species \note[Corrinne]{Stephens, 1944b Hutchinson et al., 1945, 1947 ; Gerstel and Phillips, 1958; Phillips, 1964; reviewed in Wendel and Cronn 2003}. Subsequent analyses have largely supported this observation (Abdalla et al., 2001; cronn 1999, liu 2001, Cronn et al., 1996 Seelanan et al., 1997 Small et al., 1998; Small and Wendel, 2000a,b), with few conflicts \note[Corrinne]{see, however Wendel et al., 1995b; I know there are others…. What are they?}.

A secondary outcome of this research has been the elucidation of multiple instances of hybridization among D-genome (i.e, \textit{Houzingenia}) species \note[Corrinne]{cite Cryptic Trysts}, and, in one remarkable case (i.e., \textit{G. gossypioides}), between a \textit{Houzingenia} species and another, geographically isolated subgenus from Africa (either A-, B-, or, F-genome \note[Corrinne]{cite cryptic Trysts again}). Notably, \textit{G. gossypioides} is multiply introgressant, with subsequent hybridization to a member of the \textit{G. raimondii} lineage resulting in chloroplast, if not further (and cryptic), nuclear introgression (Cronn 2003, cryptic trysts). Cytoplasmic introgression, and possibly cryptic nuclear, is also present in some populations of \textit{G. aridum}, i.e., the Mexican Colima populations; \textit{G. aridum} accessions derived from this location possess a \textit{G. davidsonii}- or \textit{G. klotzschianum}-like cytoplasm.

Modest attempts at understanding the evolution of the repetitive fraction of this genus support the inference of African introgression in \textit{G. gossypioides} \note[Corrinne]{Zhao 1998}; however, little else is understood with respect to the evolution of the non-genic fraction of \textit{Houzingenia}. The D-genome cottons possess the smallest genome sizes in the genus, ranging only ~1.11 fold, from 841 Mb – 934 Mb.  Notably, the distribution of genome sizes among the subsections suggests that this subgenus has experienced differential growth and/or reduction in genome size among species \note[Corrinne]{need a figure}; however, the patterns of sequence gain and loss have not been characterized for the subgenus. While the differences in genome size are not dramatic, there is evidence that the transposable element types which have accumulated in \textit{G. raimondii} are different than those that have achieved higher copy numbers than the remainder of the genus \note[Corrinne]{Jennifer, A domestication, kokia}. Furthermore, research comparing the two sister genera to cotton (i.e., \textit{Kokia} and \textit{Gossypioides}; \note[Corrinne]{cite Kokia paper}) reveals that their apparently static genome sizes belies both gain and loss of repetitive sequence, a result similar to that of the extant members of the A-genome (subgenus \textit{Gossypium}), whose small change in genome size (~1.05X) masks differences in element accumulation \note[Corrinne]{cite Agenome domestication}.

Modern sequencing techniques make it easy to produce a substantial amount of genomic sequencing suitable for addressing these basic questions in a more genomically comprehensive manner. Here we use modest coverage Illumina sequencing to present an in-depth view of the evolution subgenus Houzingenia, the cotton D-genome clade. We leverage newly generated genome and plastome sequences, representing the first for many species, to address questions surrounding genome evolution in a monophyletic group of closely related species. We characterize the patterns of molecular evolution of both genes and repetitive sequences to provide insight into the pace and pattern of evolution in this subgenus. For the first time, intergenic regions are evaluated to characterize the amount of divergence outside of genes, and due to indels or single-nucleotide polymorphisms (SNPs). Finally, we revisit the phylogeny of the D-genome, both adding additional insight into the relationships among species using hundreds of nuclear genes, as well as addressing questions regarding sequence gain and loss among closely related species. The genome characterized here not only provides insight into molecular evolution on a relatively recent timeframe, but it also provides resources for comparative research and the cotton community at large.

%\cite{koon,oreg,khar,zvai,xjon,schn,pond,smith,marg,hunn,advi,koha,mouse}

\section*{Results}
Text for this section \ldots
\subsection*{Results subsection}
Text for this sub-heading \ldots

\section*{Discussion}

\section*{Conclusions}

\section*{Methods}
\subsection*{Sequence generation and initial processing}
DNA was extracted from (LEAVES) using (WHAT KIT), and sent to (WHERE) for library construction and sequencing.  Sequencing was completed on the Illumina (WHAT MACHINE) using (WHICH SEQUENCING). The data were trimmed and filtered with Trimmomatic v0.32 \note[Corrinne]{citation} with the following options : (1) sequence adapter removal, (2) removal of leading and/or trailing bases when the quality score (Q) <28, (3) removal of bases after average Q <28 (8 nt window) or single base quality <10, and (4) removal of reads < 85 nt. Detailed parameters can be found at https://github.com/williamssanders/D\_Cottons\_USDA. \note[Corrinne]{Let's port this repo to a lab site after and give the new url}

\subsection*{Genome assembly and annotation}
The trimmed data was independently assembled for each species via ABySS v2.0.1 \note[Corrinne]{citation}, using every 5th kmer value from 40 through 100. A single assembly with the highest E-size  (an alternative statistic to N50; \note[Corrinne]{citation Salzberg 2011}) was selected for each species and subsequently annotated with MAKER v2.31.6 \note[Corrinne]{citation} using evidence from: (1) the NCBI \textit{G. raimondii} EST database \note[Corrinne]{citation}, (2) \textit{G. raimondii} reference genome predicted proteins, as hosted by CottonGen.org \note[Corrinne]{citation}, and (3) three \textit{ab initio} gene prediction programs, i.e. Genemark v4.30 \note[Corrinne]{citation}, SNAP v2013-11-29 \note[Corrinne]{citation}, and Augustus v3.0.3 \note[Corrinne]{citation}. Both the SNAP and Augustus models were trained using BUSCO v2.0 \note[Corrinne]{citation}.    

\subsection*{Gene stuff}
Gene orthology and family designations were determined via OrthoFinder \note[Corrinne]{citation}...

\subsection*{Repetitive characterization}
Reads from only one of the paired-end files (i.e., R1) were filtered and trimmed via Trimmomatic version 0.33 \cite{Bolger2014} to a uniform 95nt (https://github.com/williamssanders/D\_Cottons\_USDA), and then randomly subsampled to represent a 1\% genome size equivalent (GSE) for each individual \cite{Hendrix2005, Wendel2002}. These 1\% GSEs were combined as input into the RepeatExplorer pipeline \cite{Novak2013, Novak2010}, which has been successfully used to profile genomic repeats using low-coverage, short read sequencing. Only clusters which contain at least 0.01\% of the total input sequences (i.e., XXX reads from a total input of X,XXX,XXX reads) were retained for annotation as per (KOKIA CITATION), which uses the RepeatExplorer implementation of RepeatMasker \cite{Smit2015} and a custom cotton-enriched repeat library. Genome occupation of each broad repeat type was calculated (in megabases; Mb) for each genome/accession based on the 1\% genome representation of the sample and the standardized read length of 95 nt. 

Broad patterns of repeat occupation per genome were determined by using the abundance of each cluster in a multivariate dataset. Initial visualization of the data was conducted using both calculated in R \note[Corrinne]{citation} using both Principle Coordinate Analysis and Principal Component Analysis on normalized read counts, i.e., both scaled read counts and normalized by genome size for each accession. Differential abundance in cluster amounts between species was evaluated using (WHAT t-tests?), and a multivariate ANOVA was conducted using the R package \{geomorph\}. All analyses are available at (https://github.com/williamssanders/D\_Cottons\_USDA).

\subsection*{Repeat heterogeneity and relative age}
Relative cluster age was approximated using the among-read divergence profile of
each cluster, as previously used for Fritillaria \cite{Kelly2015}, dandelion
\cite{Ferreira2016}, and \textit{Kokia}/\textit{Gossypioides}, sister outgroup genera to \textit{Gossypium}. Briefly, cluster-by-cluster all-versus-all BLASTn \cite{Boratyn2013, Altshul1990} searches were conducted using the same BLAST parameters implemented in RepeatExplorer. A pairwise percent identity histogram was generated for each cluster and regression models were used to describe the trend (i.e., biased toward high-identity, “young” or lower-identity, “older” element reads) using Bayesian Information Criterion \cite{Schwarz1978} to select the model with the most confidence; specific parameters can be found in (KOKIA MANUSCRIPT) and https://github.com/williamssanders/D\_Cottons\_USDA. The read similarity profile was automatically evaluated for each cluster to determine if the reads trend toward highly similar “young” or more divergent “older” reads. These profiles generally consist of six different trends: (1) positive linear regression ("young"); (2) absence of linear
regression ("old"); (3) negative linear regression ("old"); (4) positive quadratic vertical parabola, trend described by right-side of vertex ("young"); (4b) positive quadratic vertical parabola, trend described by left-side of vertex ("old"); (5) negative quadratic vertical parabola, trend described by right-side of vertex ("old"); and (6)
negative quadratic vertical parabola, trend described by left-side of vertex and vertex at >99\% pairwise-identity ("old"; Figure WHATEVER). We note that young" and "old" are relative designations and not indicative of absolute age. 

\subsection*{Phylogenetics and ancestral state reconstruction}


%\nocite{oreg,schn,pond,smith,marg,hunn,advi,koha,mouse}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  We would like to thank....
  
  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      A short description of the figure content
      should go here.}
      \end{figure}

\begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      Figure legend text.}
      \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\begin{table}[h!]
\caption{Sample table title. This is where the description of the table should go.}
      \begin{tabular}{cccc}
        \hline
           & B1  &B2   & B3\\ \hline
        A1 & 0.1 & 0.2 & 0.3\\
        A2 & ... & ..  & .\\
        A3 & ..  & .   & .\\ \hline
      \end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
