#+TITLE: Repeat Clustering
#+DRAWERS: HIDDEN
#+OPTIONS: d:RESULTS ^:nil
#+STARTUP: hideblocks align
#+PROPERTY:  header-args :exports results :eval never-export :mkdirp yes :var DIR=(file-name-directory buffer-file-name)


Sampled reads with reservoir sort at a rate of 1% of genome size. All reads are
cut at 85 base pairs. For each accession, the run with the most trimmed forward
reads was sampled.

#+NAME: lib_table
| Accession | Genome Size (Mb) | Name                                                                                                                                                    |
|-----------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------|
| D6-7.1    |              841 | D6-7.Ggoss1.1,D6-7.Ggoss1.2                                                                                                                             |
| D6-7.2    |              841 | D6-7.Ggoss1.3,D6-7.Ggoss1.4                                                                                                                             |
| D6-7.3    |              841 | D6-7.Ggoss1.5,D6-7.Ggoss1.6                                                                                                                             |
| D1-2.2    |              841 | D1-2.Gthur1.5,D1-2.Gthur1.6                                                                                                                             |
| D1-2.3    |              841 | D1-2.Gthur1.8,D1-2.Gthur1.7                                                                                                                             |
| D6-5      |              841 | D6-5.Ggoss2.1,D6-5.Ggoss2.2,D6-5.Ggoss2.3,D6-5.Ggoss2.4                                                                                                 |
| D1-35     |              841 | D1-35.Gthur2.1,D1-35.Gthur2.2,D1-35.Gthur2.3,D1-35.Gthur2.4,D1-35.Gthur2.5                                                                              |
| D8-9      |              851 | D8-9.Gtril1.1,D8-9.Gtril1.2                                                                                                                             |
| D8-8      |              851 | D8-8.Gtril2.1                                                                                                                                           |
| D2-1-6    |              856 | D2-1-6.Garmo1.1,D2-1-6.Garmo1.2,D2-1-6.Garmo1.3,D2-1-6.Garmo1.4,D2-1-6.Garmo1.5                                                                         |
| D3-K-57   |              880 | D3-K-57.Gklot1.5,D3-K-57.Gklot1.6,D3-K-57.Gklot1.7,D3-K-57.Gklot1.8                                                                                     |
| D3K-56    |              880 | D3K-56.HFJ7YBGXX.1,D3K-56.HFJ7YBGXX.2,D3K-56.HFJ7YBGXX.3,D3K-56.HFJ7YBGXX.4,D3K-56.HHGLTBGXX.1,D3K-56.HHGLTBGXX.2,D3K-56.HHGLTBGXX.3,D3K-56.HHGLTBGXX.4 |
| D5-8.1    |              880 | D5-8.Graim1.7,D5-8.Graim1.8                                                                                                                             |
| D5-6.1    |              880 | D5-6.Graim2.1,D5-6.Graim2.2                                                                                                                             |
| D5-6.2    |              880 | D5-6.Graim2.3,D5-6.Graim2.4                                                                                                                             |
| D5-6.3    |              880 | D5-6.Graim2.5,D5-6.Graim2.6                                                                                                                             |
| D5_ref    |              880 | D5_ref                                                                                                                                                  |
| D5_2      |              880 | D5_2                                                                                                                                                    |
| D5_4      |              880 | D5_4                                                                                                                                                    |
| D5_31     |              880 | D5_31                                                                                                                                                   |
| D5_53     |              880 | D5_53                                                                                                                                                   |
| D2-2      |              910 | D2-2.HFJ7YBGXX.1,D2-2.HFJ7YBGXX.2,D2-2.HFJ7YBGXX.3,D2-2.HFJ7YBGXX.4,D2-2.HHGLTBGXX.1,D2-2.HHGLTBGXX.2,D2-2.HHGLTBGXX.3,D2-2.HHGLTBGXX.4                 |
| D10-7     |              910 | D10-7.HGF7YALXX.5                                                                                                                                       |
| D10-3     |              910 | D10-3.HG7LFALXX.1                                                                                                                                       |
| D10-8     |              910 | D10-8.HG7LFALXX.1                                                                                                                                       |
| D3-D-27   |              910 | D3-D-27.Gdavi1.1,D3-D-27.Gdavi1.2,D3-D-27.Gdavi1.3,D3-D-27.Gdavi1.4                                                                                     |
| D4-185    |              919 | D4-185.HFJ7YBGXX.1,D4-185.HFJ7YBGXX.2,D4-185.HFJ7YBGXX.3,D4-185.HFJ7YBGXX.4,D4-185.HHGLTBGXX.1,D4-185.HHGLTBGXX.2,D4-185.HHGLTBGXX.3,D4-185.HHGLTBGXX.4 |
| D4-12     |              919 | D4-12.Garid1.1,D4-12.Garid1.2,D4-12.Garid1.3,D4-12.Garid1.4                                                                                             |
| D11-1.21  |              929 | D11-1.Gschw1.1,D11-1.Gschw1.2,D11-1.Gschw1.3,D11-1.Gschw1.4,D11-1.Gschw1.7                                                                              |
| D11-1.113 |              929 | D11-1.Gschw1.5,D11-1.Gschw1.6                                                                                                                           |
| D7-157    |              934 | D7-157.HFJ7YBGXX.1,D7-157.HFJ7YBGXX.2,D7-157.HFJ7YBGXX.3,D7-157.HFJ7YBGXX.4,D7-157.HHGLTBGXX.1,D7-157.HHGLTBGXX.2,D7-157.HHGLTBGXX.3,D7-157.HHGLTBGXX.4 |
| D9-4      |              934 | D9-4.Glaxu1.1,D9-4.Glaxu1.2,D9-4.Glaxu1.3,D9-4.Glaxu1.4,D9-4.Glaxu1.5                                                                                   |
| D7-4      |              934 | D7-4.Globa1.1,D7-4.Globa1.2,D7-4.Globa1.3,D7-4.Globa1.4                                                                                                 |
| F1-1      |             1311 | F1-1                                                                                                                                                    |
| A1-97     |             1667 | A1-97                                                                                                                                                   |
| A2_4      |             1667 | A2_4                                                                                                                                                    |
| A2_34     |             1698 | A2_34                                                                                                                                                   |
| A2_1011   |             1698 | A2_1011                                                                                                                                                 |
| A1-155    |             1698 | A1-155                                                                                                                                                  |
| A2_44     |             1698 | A2_44                                                                                                                                                   |
| A2-JCVI   |             1698 | A2-JCVI                                                                                                                                                 |

Low coverage libraries
| A1-73     | 1667 | A1-73                                                           |
| D2-2-7.S3 |  910 | D2-2-7.Ghark1.1,D2-2-7.Ghark1.2,D2-2-7.Ghark1.3,D2-2-7.Ghark1.4 |
| D2-2-7.S7 |  910 | D2-2-7.Ghark1.5,D2-2-7.Ghark1.6,D2-2-7.Ghark1.7,D2-2-7.Ghark1.8 |
| D1-2.1    |  841 | D1-2.Gthur1.1,D1-2.Gthur1.2,D1-2.Gthur1.3,D1-2.Gthur1.4         |
| A2-BGI    | 1698 | A2-BGI                                                          |




* libraries seperate

#+HEADER: :shebang #!/bin/bash :tangle sample.sh :mkdirp yes
#+HEADER: :prologue #PBS -N Sample -l walltime=48:00:00 
#+BEGIN_SRC sh :var libs=lib_table
cd $DIR/
ROOT=$(git rev-parse --show-toplevel)

KEYS=("${!libs[@]}")

# Find the longest accession name and set it as the pad length
PAD='______________________________________________________'
PADLENGTH=0
for name in "${KEYS[@]}"; do
    LENGTH=$[ ${#name} ]
    if [ $PADLENGTH -lt $LENGTH ]; then PADLENGTH=$LENGTH; fi
done
PAD=${PAD:0:$PADLENGTH}

printf "Pad Length = %d\n" $PADLENGTH

# #Read through the accession table above, pad the name to the correct length, and sample reads into sampled.fa
for name in "${KEYS[@]}"; do
    readarray -t lib <<<"${libs[$name]}"
 
    LENGTH=${#name}
    NAME="${name}${PAD:$LENGTH}"
    SIZE=$[ ${lib[0]} * 10000 / 85 ]

    #Sample the largest file
    tr ',' '\n' <<<"${lib[1]}" |
        awk '{printf "%s/trim/%s.PE.R1.fq.gz\n%s/trim/%s.SE.R1.fq.gz\n", wd, $0, wd, $0}' wd=$ROOT |
        xargs echo $DIR/reservoir_sample/reservoir_sample  --read_length=85 --name="$NAME" --reservoir_size=$SIZE
#        xargs zcat | awk 'NR%4==2 {sum += length} END {print name, sum/1000/1000/size}' name=$name size=${lib[0]} OFS="\t"
done |
parallel > $DIR/sampled.libraries.fa
#+END_SRC

#+RESULTS:
: Pad Length = 9


#+HEADER: :shebang "#!/bin/bash" :mkdirp t :tangle cluster.sh
#+HEADER: :prologue #PBS -N RepeatExplorer  -l walltime=48:00:00 
#+BEGIN_SRC sh
cd $DIR
ROOT=$(git rev-parse --show-toplevel)

module load singularity-2.4.2

singularity run $DIR/repeatexplorer.squash.img -s $DIR/sampled.libraries.fa \
     -d None -v libraries -k $DIR/RBplantsANDcotton.21.08.fixedDec2016.fasta -f 9

#+END_SRC

* Libraries combined
#+NAME: acc_table
| Accession | Genome Size (Mb) | Name                                                                                                                                                    |
|-----------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------|
| A1-155    |             1698 | A1-155                                                                                                                                                  |
| A1-97     |             1667 | A1-97                                                                                                                                                   |
| A2-JCVI   |             1698 | A2-JCVI                                                                                                                                                 |
| A2_1011   |             1698 | A2_1011                                                                                                                                                 |
| A2_34     |             1698 | A2_34                                                                                                                                                   |
| A2_4      |             1667 | A2_4                                                                                                                                                    |
| A2_44     |             1698 | A2_44                                                                                                                                                   |
| D1-2      |              841 | D1-2.Gthur1.1,D1-2.Gthur1.2,D1-2.Gthur1.3,D1-2.Gthur1.4,D1-2.Gthur1.5,D1-2.Gthur1.6,D1-2.Gthur1.8,D1-2.Gthur1.7                                         |
| D1-35     |              841 | D1-35.Gthur2.1,D1-35.Gthur2.2,D1-35.Gthur2.3,D1-35.Gthur2.4,D1-35.Gthur2.5                                                                              |
| D10-3     |              910 | D10-3.HG7LFALXX.1                                                                                                                                       |
| D10-7     |              910 | D10-7.HGF7YALXX.5                                                                                                                                       |
| D10-8     |              910 | D10-8.HG7LFALXX.1                                                                                                                                       |
| D11-1     |              929 | D11-1.Gschw1.5,D11-1.Gschw1.6,D11-1.Gschw1.1,D11-1.Gschw1.2,D11-1.Gschw1.3,D11-1.Gschw1.4,D11-1.Gschw1.7                                                |
| D2-1-6    |              856 | D2-1-6.Garmo1.1,D2-1-6.Garmo1.2,D2-1-6.Garmo1.3,D2-1-6.Garmo1.4,D2-1-6.Garmo1.5                                                                         |
| D2-2      |              910 | D2-2.HFJ7YBGXX.1,D2-2.HFJ7YBGXX.2,D2-2.HFJ7YBGXX.3,D2-2.HFJ7YBGXX.4,D2-2.HHGLTBGXX.1,D2-2.HHGLTBGXX.2,D2-2.HHGLTBGXX.3,D2-2.HHGLTBGXX.4                 |
| D3-D-27   |              910 | D3-D-27.Gdavi1.1,D3-D-27.Gdavi1.2,D3-D-27.Gdavi1.3,D3-D-27.Gdavi1.4                                                                                     |
| D3-K-57   |              880 | D3-K-57.Gklot1.5,D3-K-57.Gklot1.6,D3-K-57.Gklot1.7,D3-K-57.Gklot1.8                                                                                     |
| D3K-56    |              880 | D3K-56.HFJ7YBGXX.1,D3K-56.HFJ7YBGXX.2,D3K-56.HFJ7YBGXX.3,D3K-56.HFJ7YBGXX.4,D3K-56.HHGLTBGXX.1,D3K-56.HHGLTBGXX.2,D3K-56.HHGLTBGXX.3,D3K-56.HHGLTBGXX.4 |
| D4-12     |              919 | D4-12.Garid1.1,D4-12.Garid1.2,D4-12.Garid1.3,D4-12.Garid1.4                                                                                             |
| D4-185    |              919 | D4-185.HFJ7YBGXX.1,D4-185.HFJ7YBGXX.2,D4-185.HFJ7YBGXX.3,D4-185.HFJ7YBGXX.4,D4-185.HHGLTBGXX.1,D4-185.HHGLTBGXX.2,D4-185.HHGLTBGXX.3,D4-185.HHGLTBGXX.4 |
| D5-6      |              880 | D5-6.Graim2.1,D5-6.Graim2.2,D5-6.Graim2.3,D5-6.Graim2.4,D5-6.Graim2.5,D5-6.Graim2.6                                                                     |
| D5-8      |              880 | D5-8.Graim1.7,D5-8.Graim1.8                                                                                                                             |
| D5_2      |              880 | D5_2                                                                                                                                                    |
| D5_31     |              880 | D5_31                                                                                                                                                   |
| D5_4      |              880 | D5_4                                                                                                                                                    |
| D5_53     |              880 | D5_53                                                                                                                                                   |
| D5_ref    |              880 | D5_ref                                                                                                                                                  |
| D6-5      |              841 | D6-5.Ggoss2.1,D6-5.Ggoss2.2,D6-5.Ggoss2.3,D6-5.Ggoss2.4                                                                                                 |
| D6-7      |              841 | D6-7.Ggoss1.1,D6-7.Ggoss1.2,D6-7.Ggoss1.3,D6-7.Ggoss1.4,D6-7.Ggoss1.5,D6-7.Ggoss1.6                                                                     |
| D7-157    |              934 | D7-157.HFJ7YBGXX.1,D7-157.HFJ7YBGXX.2,D7-157.HFJ7YBGXX.3,D7-157.HFJ7YBGXX.4,D7-157.HHGLTBGXX.1,D7-157.HHGLTBGXX.2,D7-157.HHGLTBGXX.3,D7-157.HHGLTBGXX.4 |
| D7-4      |              934 | D7-4.Globa1.1,D7-4.Globa1.2,D7-4.Globa1.3,D7-4.Globa1.4                                                                                                 |
| D8-8      |              851 | D8-8.Gtril2.1                                                                                                                                           |
| D8-9      |              851 | D8-9.Gtril1.1,D8-9.Gtril1.2                                                                                                                             |
| D9-4      |              934 | D9-4.Glaxu1.1,D9-4.Glaxu1.2,D9-4.Glaxu1.3,D9-4.Glaxu1.4,D9-4.Glaxu1.5                                                                                   |
| F1-1      |             1311 | F1-1                                                                                                                                                    |


#+HEADER: :shebang #!/bin/bash :tangle sample.acc.sh :mkdirp yes
#+HEADER: :prologue #PBS -N Sample -l walltime=48:00:00 
#+BEGIN_SRC sh :var libs=acc_table
cd $DIR/
ROOT=$(git rev-parse --show-toplevel)

KEYS=("${!libs[@]}")

# Find the longest accession name and set it as the pad length
PAD='______________________________________________________'
PADLENGTH=0
for name in "${KEYS[@]}"; do
    LENGTH=$[ ${#name} ]
    if [ $PADLENGTH -lt $LENGTH ]; then PADLENGTH=$LENGTH; fi
done
PAD=${PAD:0:$PADLENGTH}

printf "Pad Length = %d\n" $PADLENGTH

# #Read through the accession table above, pad the name to the correct length, and sample reads into sampled.fa
for name in "${KEYS[@]}"; do
    readarray -t lib <<<"${libs[$name]}"
 
    LENGTH=${#name}
    NAME="${name}${PAD:$LENGTH}"
    SIZE=$[ ${lib[0]} * 10000 / 85 ]

    #Sample the largest file
    tr ',' '\n' <<<"${lib[1]}" |
        awk '{printf "%s/trim/%s.PE.R1.fq.gz\n%s/trim/%s.SE.R1.fq.gz\n", wd, $0, wd, $0}' wd=$ROOT |
        xargs echo $DIR/reservoir_sample/reservoir_sample  --read_length=85 --name="$NAME" --reservoir_size=$SIZE
#        xargs zcat | awk 'NR%4==2 {sum += length} END {print name, sum/1000/1000/size}' name=$name size=${lib[0]} OFS="\t"
done |
parallel > $DIR/sampled.accessions.fa
#+END_SRC

#+RESULTS:
: Pad Length = 7


#+HEADER: :shebang "#!/bin/bash" :mkdirp t :tangle cluster.acc.sh
#+HEADER: :prologue #PBS -N RepeatExplorer  -l walltime=48:00:00 
#+BEGIN_SRC sh
cd $DIR
ROOT=$(git rev-parse --show-toplevel)

module load singularity-2.4.2

singularity run $DIR/repeatexplorer.squash.img -s $DIR/sampled.accessions.fa \
     -d None -v accessions -k $DIR/RBplantsANDcotton.21.08.fixedDec2016.fasta -f 7

#+END_SRC


* Post clustering analysis

# Cluster cutoff graph
#+BEGIN_SRC R
library(ggplot2)

data <- read.delim("analysis/clustering/seqClust/clustering/comparativeAnalysis_table_clusters_counts.csv")
data$size <- rowSums(data[,-1])
data$percent <- cumsum(data$size)/sum(data$size)

ggplot(data, aes(x=cluster, y=percent)) +
     geom_line() +
     geom_vline(xintercept=274, color='red') +
     scale_x_log10() + scale_y_log10()
ggsave('analysis/clustering/seqClust/clustering/dim_returns.cutoff.png')
#+END_SRC


#+HEADER: :shebang #!/bin/bash :tangle te_dating.sh :mkdirp yes
#+BEGIN_SRC sh 
seq -w 1 381 | parallel -j 4 \
   blastn -query accessions/seqClust/clustering/clusters/dir_CL0{}/reads.fas \
          -subject AD2.repeats.fa \
          -outfmt 6 \
          -out accessions/seqClust/clustering/clusters/dir_CL0{}/AD2.repeats.blast

find accessions/seqClust/clustering/clusters/ -name AD2.repeats.blast ! -empty |
  while read file; do 
    sed -e 's/.*dir_//' -e 's#/.*##' <<<"$file"
    cat $file | wc -l
    grep '>' $(dirname $file)/reads.fas -c 
    awk '{ if ($9 > $10){tmp=$10; $10=$9; $9=tmp}; printf "%s\t%d\t%d\n", $2, $9-1, $10}' $file | 
      sort -k1,1 -k 2,2n | 
      bedtools genomecov -max 1 -g AD2.repeats.genome -i /dev/stdin | 
      awk '$2 == 0 && $1 != "genome" {print (1-$5)*100}' OFS="\t"
  done |
  paste - - - - - |
  awk 'BEGIN{print "Cluster", "Hits", "Reads", "AF060607.1", "AF060640.1"} 1;' OFS="\t" > combineLibrary.AD2repeats.txt
#+END_SRC
